packages/terra-form-field/src/Field.jsx

```diff
@@ -131,6 +133,48 @@ const Field = (props, { intl }) => {
     labelAttrs.className,
   ]);
 
+  // Checks to run when not in production
+  if (process.env.NODE_ENV !== 'production') {
+    if (!htmlFor) {
+      // eslint-disable-next-line
+      console.warn('This prop will be required in the next major version bump of terra-form-field. It is needed for creating an accessible mapping from the form field to its related error and help text.');
+    }
+
+    if (htmlFor && hasWhiteSpace(htmlFor)) {
+      // eslint-disable-next-line
+      console.warn('The htmlFor prop should be a string without white spaces as it will be used as an HTML attribute value. Use - or _ in place of white space characters.');
+    }
+  }
+
+  let helpTextId;
+  let errorTextId;
+
+  if (htmlFor) {
+    /**
+     * IE + JAWS has trouble reading aria-describedby content with our form components.
+     * In that browser, we don't want an ID for the aria-describedby content so we have a
+     * Microsoft specific feature detect to flex on if the ID is undefined or valid.
+     */
+    helpTextId = !window.navigator.msPointerEnabled ? `${htmlFor}-help` : undefined;
```

Comment:
This looks like its going to be removed in the future. What is our next plan? https://technet.microsoft.com/en-us/windows/hh972895(v=vs.60)

Reply:
Updated this logic to be more explicit: https://github.com/cerner/terra-core/pull/2273/commits/c56e155b4d07f3cdd3b2574815aea38b2139af11